src/cata_tiles.h

```diff
@@ -25,9 +25,37 @@ class JsonObject;
 struct visibility_variables;
 
 /** Structures */
+struct sprite_variation {
+    int weight;
+    std::vector<int> sprite_ids;
+
+    sprite_variation()
+    {
+        weight = 1;
```

Comment:
the default weight of 1 is functional for every case except the first one. on line 723 I'm creating an "invalid" entry for storing the O(1) lookup list of other variations. every other instance of a sprite_variation uses a positive weight, and most of them use the default weight of 1.